inc/Logger.hpp

```diff
@@ -33,11 +33,12 @@ class Logger {
 private:
 	// private constructor to prevent direct object creation
 	Logger();
+	// private destructor to prevent direct object deletion
+	~Logger();
```

Comment:
As you're not performing any manual tasks in the destructor, you can remove this declaration and the definition here and just use the implicit default destructor.

Reply:
Logger is a singleton. Nobody should be able to call the destructor of Logger (// private destructor to prevent direct object deletion).

If it was public, you could call the destructor anywhere (as it's just another function) leading to crash.

Reply:
You mean if the declaration as private is removed and the default destructor is used, this destructor would be public and accessible from anyone?